src/LocalDatastoreController.localStorage.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+/* global localStorage */
+
+const LocalDatastoreController = {
+  fromPinWithName(name: string): ?any {
+    const values = localStorage.getItem(name);
+    if (!values) {
+      return null;
+    }
+    const objects = JSON.parse(values);
+    return objects;
+  },
+
+  pinWithName(name: string, value: any) {
+    try {
+      const values = JSON.stringify(value);
+      localStorage.setItem(name, values);
+    } catch (e) {
+      // Quota exceeded, possibly due to Safari Private Browsing mode
```

Comment:
we should log something still.

Thie is important for the dev to know at dev time that the local storage exploded for some reason :)

Reply:
What text should be logged? There are a lot of potential reasons why this would fail

Reply:
The error itself